projects/distributed-tracing/src/shared/dashboard/widgets/log-detail/data/log-detail-data-source.model.ts

```diff
@@ -0,0 +1,21 @@
+import { Dictionary } from '@hypertrace/common';
+import { Model, ModelProperty, PLAIN_OBJECT_PROPERTY } from '@hypertrace/hyperdash';
+import { Observable, of } from 'rxjs';
+import { GraphQlDataSourceModel } from '../../../data/graphql/graphql-data-source.model';
+import { LogEvent } from '../../waterfall/waterfall/waterfall-chart';
+
+@Model({
+  type: 'log-detail-data-source'
```

Comment:
Not sure this page really needs to be a widget - I wouldn't actively avoid it, and if the work is already done it's fine - but for future pages, we really need to get out of the habit of making complex full-screen widgets.

Reply:
I think it makes sense for this to be a widget so that it can go in as a full page list in the trace level view in Phase 2.

Reply:
I'd extend the same argument to those views too though 😉 If I understand the comment/remember the screen, the trace view today is one big widget, and with this it will now have two views - the sequence and the log list, right? When we do that work, I'd make the same argument that the tabs and their contents need not be (but easily could be) dashboards.

Reply:
@itssharmasandeep This is for the tab view right? don't we fetch all the details beforehand and show the tab as a normal component?

Reply:
Yeah, but since we're using table dashboard, so for child template we need data source, right?
Currently I am not fetching anything.

Reply:
Hmm I think my comment got lost. My original comment was about the table itself - if that needed to be a dashboard (and likewise, everything downstream). The dashboards have really... evolved past how they were intended to be used. Here for example, we could just as easily remove the data source and pass the data directly to the widget - it's not providing any value if it's a custom pass through for a custom widget. (just to clarify, we don't need to change this here, but it's something we should be cognizant of as we build new screens).

As far as this data source goes, can we just reuse `StaticDataSource` since this one's not doing anything but unwrapping which we could do elsewhere? Or if we keep it, it shouldn't extend `GraphQlDataSourceModel` because it's not querying GQL.

Reply:
+1 to aaron's point. 

@itssharmasandeep For the sheet/tab view, we don't necessarily need a data source/widget for child template. You can just pass any ng-template in it and it would render. For the page view, we can use the static data source like Aaron described above.

Reply:
Shared the details on slack